File: Clubmanagement/src/main/java/com/example/Clubmanagement/entities/compte/generlAc/Compte.java
package com.example.Clubmanagement.entities.compte.generlAc;


import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Collection;


@Entity
@Table
@Data
@NoArgsConstructor
@RequiredArgsConstructor

public class Compte {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id" ,nullable = false)

    private Long idE;
    @Column(nullable = false)
    private String civilite ;
    @Column(nullable = false)
    private String fullname;//Nom et prenom
    @Column(nullable = false)
    private  String email;
    @Column(nullable = false)
    private String pass;
    @Column(nullable = false)
    private Long tel;
    @ManyToMany(fetch = FetchType.EAGER)
   private  Collection<Role> roles=new ArrayList<>();


    public Compte(String civilite, String fullname, String email, String pass, Long tel) {
        this.civilite = civilite;
        this.fullname = fullname;
        this.email = email;
        this.pass = pass;
        this.tel = tel;
    }

    public Compte(Long id, String civilite, String fullname, String email, String pass, Long tel) {
        this.idE = id;
        this.civilite = civilite;
        this.fullname = fullname;
        this.email = email;
        this.pass = pass;
        this.tel = tel;
    }


    //end constructor


    public Long getId() {
        return idE;
    }

    public void setId(Long id) {
        this.idE = id;
    }

    public String getCivilite() {
        return civilite;
    }

    public void setCivilite(String civilite) {
        this.civilite = civilite;
    }

    public String getfullname() {
        return fullname;
    }

    public void setfullname(String fullname) {
        this.fullname = fullname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPass() {
        return pass;
    }

    public void setPass(String pass) {
        this.pass = pass;
    }

    public Long getTel() {
        return tel;
    }

    public void setTel(Long tel) {
        this.tel = tel;
    }


    @Override
    public String toString() {
        return "Compte{"+
                " civilite='" + civilite + '\'' +
                ", fullname='" + fullname + '\'' +
                ", email='" + email + '\'' +
                ", pass='" + pass + '\'' +
                ", tel=" + tel +
                '}';
    }








}

File: Clubmanagement/src/front-end/Classes/compte.ts
export class compte {

    private _idE:number;
    private _civilite:number;
    private _fullname:String;
     private _email: String;
     private _pass :String;
     private _tel:number;
     private _roles: Array<String>;


     get idE(): number {
         return this._idE;
     }

     set idE(value: number) {
         this._idE = value;
     }


    get civilite(): number {
        return this._civilite;
    }

    set civilite(value: number) {
        this._civilite = value;
    }

    get fullname(): String {
        return this._fullname;
    }

    set fullname(value: String) {
        this._fullname = value;
    }

    get email(): String {
        return this._email;
    }

    set email(value: String) {
        this._email = value;
    }


    get pass(): String {
        return this._pass;
    }

    set pass(value: String) {
        this._pass = value;
    }

    get tel(): number {
        return this._tel;
    }

    set tel(value: number) {
        this._tel = value;
    }

    get roles(): Array<String> {
        return this._roles;
    }

    set roles(value: Array<String>) {
        this._roles = value;
    }

    constructor(idE: number, civilite: number, fullname: String, email: String, password: String, tel: number) {
         this._idE = idE;
         this._civilite = civilite;
         this._fullname = fullname;
         this._email = email;
         this._pass = password;
         this._tel = tel;
     }
 }